aea/agent_loop.py

```diff
@@ -172,59 +174,69 @@ def _behaviour_exception_callback(self, fn: Callable, exc: Exception) -> None:
         :return: None
         """
         self.logger.exception(
-            f"Loop: Exception: `{exc}` occured during `{fn}` processing"
+            f"Loop: Exception: `{exc}` occured during `{task_callable}` processing"
         )
         self._exceptions.append(exc)
         self._state.set(AgentLoopStates.error)
 
-    def _register_behaviour(self, behaviour: Behaviour) -> None:
+    def _register_periodic_task(
+        self,
+        task_callable: Callable,
+        period: float,
+        start_at: Optional[datetime.datetime],
+    ) -> None:
         """
         Register behaviour to run periodically.
 
         :param behaviour: Behaviour object
 
         :return: None
         """
-        if behaviour in self._behaviours_registry:  # pragma: nocover
+        if task_callable in self._periodic_tasks:  # pragma: nocover
             # already registered
             return
 
         periodic_caller = PeriodicCaller(
             partial(
                 self._agent._execution_control,  # pylint: disable=protected-access # TODO: refactoring!
-                behaviour.act_wrapper,
-                behaviour,
+                task_callable,
+                task_callable,
             ),
-            behaviour.tick_interval,
-            behaviour.start_at,
-            self._behaviour_exception_callback,
-            self._loop,
+            period=period,
+            start_at=start_at,
+            exception_callback=self._periodic_task_exception_callback,
+            loop=self._loop,
         )
-        self._behaviours_registry[behaviour] = periodic_caller
+        self._periodic_tasks[task_callable] = periodic_caller
         periodic_caller.start()
-        self.logger.debug(f"Behaviour {behaviour} registered.")
+        self.logger.debug(f"Periodic task {task_callable} registered.")
 
-    def _register_all_behaviours(self) -> None:
+    def _register_periodic_tasks(self) -> None:
         """Register all AEA behaviours to run periodically."""
-        for behaviour in self._agent.active_behaviours:
-            self._register_behaviour(behaviour)
-
-    def _unregister_behaviour(self, behaviour: Behaviour) -> None:
+        for (
+            task_callable,
+            (period, start_at),
+        ) in (
+            self._agent._get_behaviours_tasks().items()  # pylint: disable=protected-access
```

Comment:
should not be calling `_get_behaviours_tasks()`...as this is AEA specific...should be `_get_periodic_tasks()` I believe

Reply:
Also; it's clear that `_get_periodic_tasks` should be public not private, please change

Reply:
the idea was that async main loop is not supported by agent only by aea, but after refactoring, i missed this check.
so your notice is helpful. anyway, it will be refactored after a while.

Reply:
done